card/mod/06_bootstrap/set/all/bootstrap/form.rb

```diff
@@ -1,15 +1,14 @@
 format :html do
   def button_tag content_or_options = nil, options = {}, &block
-    if block_given?
-      content_or_options[:class] ||= ''
-      content_or_options[:class] += ' btn btn-default'
-    else
-      options[:class] ||= ''
-      options[:class] += ' btn btn-default'
-    end
+    bootstrapify_button( block_given? ? content_or_options : options )
     super(content_or_options, options, &block)
   end
   
+  def bootstrapify_button options
+    situation = options.delete(:situation) || 'default'
+    options[:class] = [ options[:class], 'btn', "btn-#{situation}" ].compact*' '
+  end
```

Comment:
not sure "situation" is perfect, but "context" already has lots of meaning in wagn, and "state" seems wrong too (those are the two bootstrap words for this).

default, primary, info, success, warning, danger.... hard to find a word that feels really natural for all of those.